### PR TITLE
Add Cockpit to Easypanel 🎉 

### PR DESCRIPTION
**Pull request by Supernova3339**

### CODE REVIEW

**Comment by Ponkhy** [comment]
Hey, is this template for [Cockpit-HQ](https://hub.docker.com/r/cockpithq/cockpit) or the old v1 [Cockpit](https://hub.docker.com/r/agentejo/cockpit/)?

**Comment by Ponkhy** [comment]
> brand new v2 - built a custom image for it on easypanel-community/docker-cockpit-cms

What is the difference between the custom and official v2 image?

**Comment by Supernova3339** [comment]
> > brand new v2 - built a custom image for it on easypanel-community/docker-cockpit-cms
> 
> What is the difference between the custom and official v2 image?

there isn't a v2 image - only v1. v2 doesn't go up to the latest version, it goes up to 2.11.5

**Comment by Ponkhy** [comment]
> > > brand new v2 - built a custom image for it on easypanel-community/docker-cockpit-cms
> > 
> > 
> > What is the difference between the custom and official v2 image?
> 
> there isn't a v2 image - only v1. v2 doesn't go up to the latest version, it goes up to 2.11.5

Wouldn't this image [core-2.4.1](https://hub.docker.com/layers/cockpithq/cockpit/core-2.4.1/images/sha256-b58539d5244b3f6ed07368a39ba606e8a887f1400e8e1f1b4a024cebed2dea69?context=explore) be the one for the latest version [2.4.1](https://github.com/Cockpit-HQ/Cockpit/releases/tag/2.4.1), or am I mistaken?

**Comment by Supernova3339** [comment]
> > > > brand new v2 - built a custom image for it on easypanel-community/docker-cockpit-cms
> > > 
> > > 
> > > What is the difference between the custom and official v2 image?
> > 
> > 
> > there isn't a v2 image - only v1. v2 doesn't go up to the latest version, it goes up to 2.11.5
> 
> Wouldn't this image [core-2.4.1](https://hub.docker.com/layers/cockpithq/cockpit/core-2.4.1/images/sha256-b58539d5244b3f6ed07368a39ba606e8a887f1400e8e1f1b4a024cebed2dea69?context=explore) be the one for the latest version [2.4.1](https://github.com/Cockpit-HQ/Cockpit/releases/tag/2.4.1), or am I mistaken?

its for 2.4.1

File: templates/cockpit/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      },
+      proxy: {
+        port: 80,
+        secure: true,
+      },
+      mounts: [
+        {
+          type: "volume",
+          name: "html",
+          mountPath: "/var/ww/html"
```
Comment (Ponkhy) [inline templates/cockpit/index.ts:32]:
```suggestion
          mountPath: "/var/www/html"
```

**Review by Ponkhy — CHANGES_REQUESTED** [review]
The data doesn't seem to be persistent after a restart, at least the login. Could you please look into it?

**Comment by Supernova3339** [comment]
could you attempt deployment to confirm? I've just tried and had no
problems.

**Comment by Ponkhy** [comment]
> could you attempt deployment to confirm? I've just tried and had no problems.

Yes I can confirm it, you can try it yourself by just stopping the app and start it again, after that you'll not be able to login again.
Although I think the MongoDB connections string is also wrong, or is not even used.

File: templates/cockpit/index.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  services.push({
+    type: "app",
+    data: {
+      projectName: input.projectName,
+      serviceName: input.appServiceName,
+      env: [
+        `COCKPIT_SESSION_NAME=cockpit`,
+        `COCKPIT_SALT=${databaseSalt}`,
+        `COCKPIT_DATABASE_SERVER=mongodb://mongo:${mongoPassword}:27017`,
+        `COCKPIT_DATABASE_NAME=${input.projectName}_${input.databaseServiceName}`
```
Comment (Ponkhy) [inline templates/cockpit/index.ts:18]:
```suggestion
        `COCKPIT_DATABASE_SERVER=mongodb://mongo:${mongoPassword}@${input.projectName}_${input.databaseServiceName}:27017`,
        `COCKPIT_DATABASE_NAME=${input.projectName}`
```

**Review by Ponkhy — CHANGES_REQUESTED** [review]
It seems to work now, although I don't think that it actually needs/uses the MongoDB.

**Pull request merged**